include/proxy-wasm/compat.h

```diff
@@ -14,28 +14,33 @@
 
 #pragma once
 
-// #ifdef PROXY_WASM_USE_ABSL
-#ifndef __cpp_lib_optional
-#include "absl/types/optional.h"
-#include "absl/strings/string_view.h"
-#else
+// Provide compatibiliby for projects which have not yet moved to C++17.
+// TODO: remove this when all dependent projects have upgraded.
```

Comment:
What's the benefit of not keeping this compatibility going forward? Seems like it would be low-maintenance.

Reply:
Just makes things a bit messy.